pkg/api/auth.go

```diff
@@ -0,0 +1,89 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	log "github.com/els0r/log"
+)
+
+func (k APIKeys) Exists(key string) bool {
+	_, exists := k[key]
+	return exists
+}
+
+func checkKeyConstraints(key string) error {
+	// enforce long API keys (e.g. SHA256)
+	if len(key) < 32 {
+		return fmt.Errorf("key considered insecure: insufficient key length %d", len(key))
+	}
+
+	// TODO: consider to check entropy of key
```

Comment:
Cool idea! However, then you probably couldn't call it on each auth request (maybe too expensive)...

Reply:
Yeah. It could be done as a check during config loading rather than for each request. Same with key length checks.